### PR TITLE
feat: add telemetry for webview

### PR DESCRIPTION
**Pull request by HuihuiWu-Microsoft**
Add telemetry for webview actions

### CODE REVIEW

File: packages/vscode-extension/src/controls/Commands.ts [hunk 1]
```diff
@@ -4,7 +4,6 @@
 export enum Commands {
   OpenExternalLink = "open-external-link",
   CloneSampleApp = "clone-sample-app",
-  DisplayCommandPalette = "display-command-palette",
```
Comment (HuihuiWu-Microsoft) [inline packages/vscode-extension/src/controls/Commands.ts:7]: Clean unused codes

File: packages/vscode-extension/src/controls/QuickStart.tsx [hunk 2]
```diff
@@ -270,7 +275,7 @@ export default class QuickStart extends React.Component<any, any> {
                   className="capabilitiesVideo"
                   controls
                   disablePictureInPicture
-                  onPlay={this.onWatchVideo}
+                  onPlay={this.onVideoPlay}
```
Comment (HuihuiWu-Microsoft) [inline packages/vscode-extension/src/controls/QuickStart.tsx:278]: To call onWatchVideo() here will send telemetry twice, so separate play action with following actions

File: packages/vscode-extension/src/controls/webviewPanel.ts [hunk 3]
```diff
@@ -161,6 +120,68 @@ export class WebviewPanel {
...
+          } else {
+            vscode.window.showErrorMessage("Failed to download sample app");
+          }
+        } finally {
+          progress.end();
+        }
+      }
+    }
+
+    if (downloadSuccess) {
```
Comment (HuihuiWu-Microsoft) [inline packages/vscode-extension/src/controls/webviewPanel.ts:177]: Send the app name property when download sample app

File: packages/vscode-extension/src/controls/QuickStart.tsx [hunk 4]
```diff
@@ -319,13 +324,22 @@ export default class QuickStart extends React.Component<any, any> {
     }
   };
 
-  onNextStep = (step: number) => {
+  onNextStep = (step: number, title: string) => {
+    vscode.postMessage({
```
Comment (HuihuiWu-Microsoft) [inline packages/vscode-extension/src/controls/QuickStart.tsx:328]: Send current step(both number and step name) property when user click next

**Review by nliu-ms — APPROVED** [review]

**Review by LongOddCode — APPROVED** [review]

File: packages/vscode-extension/src/controls/webviewPanel.ts [hunk 5]
```diff
@@ -161,6 +120,61 @@ export class WebviewPanel {
...
+      ExtTelemetry.sendTelemetryEvent(TelemetryEvent.DownloadSample, { [TelemetryProperty.TriggerFrom]: TelemetryTiggerFrom.Webview, [TelemetryProperty.SampleAppName]: msg.data.appFolder, [TelemetryProperty.Success]: TelemetrySuccess.Yes });
+    } else {
+      ExtTelemetry.sendTelemetryEvent(TelemetryEvent.DownloadSample, { [TelemetryProperty.TriggerFrom]: TelemetryTiggerFrom.Webview, [TelemetryProperty.SampleAppName]: msg.data.appFolder, [TelemetryProperty.Success]: TelemetrySuccess.No });
```
Comment (chagong) [inline packages/vscode-extension/src/controls/webviewPanel.ts:174]: I think here you need to sendTelemetryErrorEvent. and compose an error as parameter to it.

**Review by chagong — APPROVED** [review]

**Pull request merged**